### ISSUE TITLE
schema/gen/go: could probably be simpler if we ran gofmt on the output

### ISSUE THREAD

**Issue by mvdan** [ipld/go-ipld-prime#103]
Right now, the gen/go code is very careful to produce code that complies with gofmt, from what I can see.

I think this is a good idea, but it's also more trouble than it's worth. It requires extra code complexity, and gofmt is a moving target since it can change between Go releases.

Instead, I propose that we simplify the code and just use https://golang.org/pkg/go/format/ on the produced files before writing them to disk. This would add an extra `[]byte` pass and could slow down the generator slightly, but I doubt that would be noticeable or even noteworthy, since this is not a generator that one would run often.

**Comment by warpfork** [comment]
(For the record, it's not that careful.  There may be aspirational comments, and I try to get things "close", but... lol.)

Calling `go/format` programmatically rather than leaving it as a problem for an end user to script sounds good, and we should do it :+1: 

It would be kinda nice if we can keep the interfaces overall such that most of the generation system doesn't know about filenames, but it sounds like that should still be possible.

**Comment by mvdan** [comment]
Yep, formatting is done either at the AST or []byte level, both of which can be independent of filenames and files.

**Comment by mvdan** [comment]
In a similar way, I've found a lot of the template code hard to read, because the template blocks/logic aren't indented. It's then hard to see what the corresponding `{{ range }}` or `{{ if }}` is to an `{{ end }}` line.

If we simply rely on go/format to produce Go formatting, we could then indent our templates to be easier to read and maintain at the same time.
